src/Provider/TaxonUrlProvider.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Menu plugin for Sylius.
+ *
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusMenuPlugin\Provider;
+
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class TaxonUrlProvider extends AbstractUrlProvider
+{
+    public const PROVIDER_CODE = 'taxon';
+
+    protected string $code = self::PROVIDER_CODE;
+
+    protected string $icon = 'folder';
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** ‏ Really not fan of the icon folder when selecting a taxon because it looks like we will not to choose something else after (like a folder in a file manager)

Reply:
It's the same icon as Sylius in menu bar :/

Reply:
yeah I know